packages/server/modules/core/rest/diffUpload.js

```diff
@@ -25,19 +25,22 @@ module.exports = (app) => {
     }
 
     const objectList = JSON.parse(req.body.objects)
-    if (objectList.length > MAXIMUM_OBJECTS) {
-      req.log.warn(
-        `User ${req.context.userId} tried to diff ${objectList.length} objects, which is greater than the maximum of ${MAXIMUM_OBJECTS}.`
-      )
-      return res.status(400).end(`Too many objects. Maximum ${MAXIMUM_OBJECTS}.`)
-    }
 
     req.log.info(`Diffing ${objectList.length} objects.`)
 
-    const response = await hasObjects({
-      streamId: req.params.streamId,
-      objectIds: objectList
-    })
+    const chunkSize = 65536 // maximum size of unsigned 16 bit integer
```

Comment:
Let's lower this to a more reasonable number. 
With the split promises we can hammer the DB a bit more. 

It will also help reducing the effects of the suboptimal implementation of the hasObjects function.

I think we can start with 500 or 1000.